test/stateful_dataloader/test_state_dict.py

```diff
@@ -1491,5 +1491,64 @@ def test_pw(self):
         self._run(True, 4)
 
 
+class CountIterCalls(torch.utils.data.IterableDataset):
```

Comment:
Would it be possible to add another iterable dataset which returns itself as the iterator (similar to DummyIteratorIterableDataset) so that flow is also tested?